test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/CreateMockitoMocksCallback.java

```diff
@@ -25,30 +28,41 @@ public void afterConstruct(Object testInstance) {
                 InjectMock injectMockAnnotation = field.getAnnotation(InjectMock.class);
                 if (injectMockAnnotation != null) {
                     Object beanInstance = getBeanInstance(testInstance, field, InjectMock.class);
-                    Object mock = createMockAndSetTestField(testInstance, field, beanInstance);
-                    MockitoMocksTracker.track(testInstance, mock, beanInstance);
+                    Map.Entry<Boolean, Object> result = createMockAndSetTestField(testInstance, field, beanInstance);
+                    if (result.getKey()) {
+                        MockitoMocksTracker.track(testInstance, result.getValue(), beanInstance);
+                    }
                 }
             }
             current = current.getSuperclass();
         }
     }
 
-    private Object createMockAndSetTestField(Object testInstance, Field field, Object beanInstance) {
+    private Map.Entry<Boolean, Object> createMockAndSetTestField(Object testInstance, Field field, Object beanInstance) {
```

Comment:
Hm, why the `Map.Entry` here? Something like an `Optional<Object>` would be probably more clear, or?

Reply:
Yeah, I guess you are right. I'll fixi t

Reply:
Fixed